src/apps/backend/modules/task/internal/task-writer.ts

```diff
@@ -61,4 +63,30 @@ export default class TaskWriter {
       },
     );
   }
+
+  public static async shareTask(params: ShareTaskParams): Promise<Task> {
+    const userIds = params.userIds.map((id) => new Types.ObjectId(id));
+
+    const task = await TaskRepository.findOneAndUpdate(
+      {
+        account: params.accountId,
+        _id: params.taskId,
+        active: true,
+      },
+      {
+        $addToSet: {
+          sharedWith: { $each: userIds },
+        },
+      },
+      { new: true },
+    )
+      .lean()
+      .exec();
+
+    if (!task) {
+      throw new TaskNotFoundError(params.taskId);
+    }
+
+    return TaskUtil.convertTaskDBToTask(task as TaskDB);
+  }
```

Comment:
we are just updating the `sharedWith` field. So instead of creating a separate method for this we can do this in the the existing `updateTask` method like


```
{
  $set: {
    description: params.description,
    title: params.title,
  },
  $addToSet: {
    sharedWith: { $each: userIds },
  },
},
```